incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -114,6 +116,9 @@ func (v *Hierarchy) handle(ctx context.Context, log logr.Logger, req *request) a
 	return v.checkAuthz(ctx, req.ui, authzReqs)
 }
 
+const validRCRexFmt string = "^[a-z0-9]([-a-z0-9]*[a-z0-9])?[a-z0-9]$"
+const validRCMaxLength int = 63
+
```

Comment:
Can you please comment:

- What the names mean ("RC?")
- Where you got these regexes from (i.e., if you didn't cut-and-paste the code, where's the documentation that told you how to write them)?

Also, what you're really doing here is validating that the namespace name is legal. Did you check whether any Kubernetes golang package can do this check for you?

Reply:
- Required Child (RC)

-When creating a namespace with invalid character, it will return info about regex used for validation.i referenced from there and modify it since it's not working when I implement and test it. I found the "real" regexes from kubernetes package. I will change according to it. 

- There is a kubernetes package for validating name but the function name (validation.IsDNS1123Label) seems not suitable here IMO. Should I go with that one?

Reply:
Oh, I wasn't asking for you to comment _here_ - I meant you should add a comment to the _code_, using [godoc](https://blog.golang.org/godoc-documenting-go-code) style. Eg:

```
// validRCRexFmt is the regex used to validate namespaces for the required children
// field in the hierarchy spec. It's based on this: <link>
```

However, I _would_ use `validation.IsDNS1123Label` if that's what K8s uses, and just leave a comment in the code _why_ you did that.